examples/hello_header/src/main.rs

```diff
@@ -0,0 +1,67 @@
+//! A simple introduction to working with Gotham and custom headers.
+
+extern crate futures;
+extern crate gotham;
+#[macro_use]
+extern crate hyper;
+extern crate mime;
+
+use hyper::{Response, StatusCode};
+use gotham::http::response::create_response;
+use gotham::state::State;
+use gotham::router::Router;
+use gotham::router::builder::{build_simple_router, DefineSingleRoute, DrawRoutes};
+
+// this macro defines our custom header
+header! { (GothamHeader, "X-Gotham") => [String] }
+
+/// Create a `Handler` that adds a custom header.
+pub fn say_hello_through_header(state: State) -> (State, Response) {
+    let mut res = create_response(
+        &state,
+        StatusCode::Ok,
+        Some((b"".to_vec(), mime::TEXT_PLAIN)),
```

Comment:
oops I missed this earlier, all this needs to be is a `None` as you don't want/need a body + mime type in this response.